src/main/java/com/salesforce/dataloader/controller/Controller.java

```diff
@@ -225,7 +234,7 @@ public void createAndShowGUI() throws ControllerInitializationException {
             String errMsg = Messages.getFormattedString("Controller.errorConfigWritable", config.getFilename());
 
             String currentWorkingDir = System.getProperty("user.dir");
-            if (currentWorkingDir.startsWith("/Volumes")){
+            if (currentWorkingDir.startsWith("/Volumes")) {
```

Comment:
The working dir is not the install dir; this will not reliably catch the user running from a DMG (you'd need to check the install dir for that. See `getInstalledConfigDir` for an example.